### PR TITLE
Resource Navigation Link and Service Association Link merge issues

### PR DESCRIPTION
**Pull request by wdehrich**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Review by MikhailTryakhov — APPROVED** [review]

File: specification/network/resource-manager/Microsoft.Network/stable/2019-04-01/virtualNetwork.json
Comment (dsgouda) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-04-01/virtualNetwork.json]: Please mark this as readonly

File: specification/network/resource-manager/Microsoft.Network/stable/2019-04-01/virtualNetwork.json
Comment (dsgouda) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-04-01/virtualNetwork.json]: Please mark this as readonly

**Review by dsgouda — CHANGES_REQUESTED** [review]
Looks good for the most part

File: specification/network/resource-manager/Microsoft.Network/stable/2019-04-01/virtualNetwork.json
Comment (wdehrich) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-04-01/virtualNetwork.json]: fixed

File: specification/network/resource-manager/Microsoft.Network/stable/2019-04-01/virtualNetwork.json
Comment (wdehrich) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-04-01/virtualNetwork.json]: fixed

**Review by dsgouda — APPROVED** [review]
LGTM will merge once CIs pass

**Comment by MikhailTryakhov** [comment]
@dsgouda can you please merge it once it's approved by Gaurav?

**Comment by MikhailTryakhov** [comment]
@dsgouda can you please merge it because all the jobs except the one which failed are about to pass...

**Comment by EvgenyAgafonchikov** [comment]
amended the last commit with no changes to retrigger CI validations

**Pull request merged**
